### ISSUE TITLE
Windows triton build cannot make https call because of a reduced default libcurl 

### ISSUE THREAD

**Issue by RandySheriffH** [triton-inference-server/server#5000]
With either r20.03 or main branch, after build with triton client on windows and coded an exe base off cxx API:
The line where inference is happening failed:

```
std::unique_ptr<tc::InferenceServerHttpClient> client;
if (!tc::InferenceServerHttpClient::Create(&client, "https://abc.com", true).IsOk()) {
	throw std::exception("Failed to create triton client!");
}
...
if(!client->Infer(&results, options, triton_inputs, triton_outputs, http_headers, tc::Parameters(),
                          request_compression_algorithm, response_compression_algorithm).IsOk()) {
			throw std::exception("Failed to send triton http request!");										
}
``` 

libcurl reports:

```
* Protocol "https" not supported or disabled in libcurl
* Closing connection -1
```

The root cause of this is that my exe linked to libcurl.lib in triton client: 

```
c:/triton/client/build/third-party/curl
```

This pre-built  libcurl is compiled with zero ssl backend support such as openssl or schannel.
So I git-cloned curl and built with:

```
cmake .. -DCURL_USE_SCHANNEL=ON
```

After relinking, the https call succeed. 
To sum up, I am expecting either 1 or 2:

1.  document that customer might need to rebuild customized libcurl with ssl properly enabled;
2.  when building on specific platform, enable ssl when building the default libcurl.

**Comment by GuanLuo** [comment]
@jbkyang-nvi @nv-kmcgill53 Looks like we do [build and use curl library](https://github.com/triton-inference-server/third_party/blob/main/CMakeLists.txt#L74-L95) as part of the build process, I think we need to make sure we build it with SSL support

**Comment by kthui** [comment]
Hi, the fix for building libcurl with winssl is merged into third_party repo, and the fix for compiling the http_client.dll is merged into client repo. Please feel free to comment/re-open this ticket if you still experience issue with libcurl/compilation. Thanks for your patience.
